third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1340,501 +1341,1256 @@ pub struct CodeUnit {
+    "#]
     VecSwap(SignatureIndex),
-    /// Push a U16 constant onto the stack.
-    ///
-    /// Stack transition:
-    ///
-    /// ```... -> ..., u16_value```
+
+    #[group = "stack_and_local"]
+    #[description = "Push a u16 constant onto the stack."]
+    #[static_operands = "[u16_value]"]
+    #[semantics = "stack << u16_value"]
+    #[paranoid_post = "ty_stack << u16"]
     LdU16(u16),
-    /// Push a U32 constant onto the stack.
-    ///
-    /// Stack transition:
-    ///
-    /// ```... -> ..., u32_value```
+
+    #[group = "stack_and_local"]
+    #[description = "Push a u32 constant onto the stack."]
+    #[static_operands = "[u32_value]"]
+    #[semantics = "stack << u32_value"]
+    #[paranoid_post = "ty_stack << u32"]
     LdU32(u32),
-    /// Push a U256 constant onto the stack.
-    ///
-    /// Stack transition:
-    ///
-    /// ```... -> ..., u256_value```
+
+    #[group = "stack_and_local"]
+    #[description = "Push a u256 constant onto the stack."]
+    #[static_operands = "[u256_value]"]
+    #[semantics = "stack << u256_value"]
+    #[paranoid_post = "ty_stack << u256"]
     LdU256(move_core_types::u256::U256),
-    /// Convert the value at the top of the stack into u16.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., integer_value -> ..., u16_value```
+
+    #[group = "casting"]
+    #[description = r#"
+        Convert the integer value at the top of the stack into a u16. 
+        An arithmetic error will be raised if the value cannot be represented as a u16.
+    "#]
+    #[semantics = r#"
+        stack >> int_val
+        if int_val > u16::MAX:
+            arithmetic error
+        else:
+            stack << int_val as u16
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> _
+        ty_stack << u16
+    "#]
     CastU16,
-    /// Convert the value at the top of the stack into u32.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., integer_value -> ..., u32_value```
+
+    #[group = "casting"]
+    #[description = r#"
+        Convert the integer value at the top of the stack into a u32. 
+        An arithmetic error will be raised if the value cannot be represented as a u32.
+    "#]
+    #[semantics = r#"
+        stack >> int_val
+        if int_val > u32::MAX:
+            arithmetic error
+        else:
+            stack << int_val as u32
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> _
+        ty_stack << u32
+    "#]
     CastU32,
-    /// Convert the value at the top of the stack into u256.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., integer_value -> ..., u256_value```
+
+    #[group = "casting"]
+    #[description = r#"
+        Convert the integer value at the top of the stack into a u256.
+    "#]
+    #[semantics = r#"
+        stack >> int_val
+        stack << int_val as u256
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> _
+        ty_stack << u256
+    "#]
     CastU256,
 }
 
+#[test]
+fn tttt() {
```

Comment:
Maybe a nicer name? :)

Reply:
My bad, used this as an ad-hoc entry point to print out the extracted spec. Will get it removed.